sympy/matrices/matrices.py

```diff
@@ -3433,6 +3433,99 @@ def _eval_matrix_exp_jblock(self):
         from .sparsetools import banded
         return self.__class__(banded(size, bands))
 
+
+
+    def analytic_func(self, f, x, parameter=None):
+        """
+        Computes f(A) where A is a Square Matrix
+        and f is an analytic function.
+
+        Examples
+        ========
+
+        >>> from sympy import Symbol, Matrix, exp, S, log
+
+        >>> x = Symbol('x')
+        >>> m = Matrix([[S(5)/4, S(3)/4], [S(3)/4, S(5)/4]])
+        >>> f = log(x)
+        >>> m.analytic_func(f, x)
+        Matrix([
+        [     0, log(2)],
+        [log(2),      0]])
```

Comment:
I excerpt an another test example from this
```
>>> from sympy import *
>>> x = Symbol('x')
>>> t = Symbol('t')
>>> m = Matrix([[S(5)/4, S(3)/4], [S(3)/4, S(5)/4]])
>>> m.analytic_func(log(t), x, t)
Matrix([
[log(t),      0],
[     0, log(t)]])
>>> _.exp()
Matrix([
[t, 0],
[0, t]])
```

But it doesn't look right.

Reply:
make it m.analytic_func(log(x), x, t)

Reply:
Okay, I'd got it

Reply:
i should probably simplify the expression before returning?

Reply:
Unfortunately, simplifying a matrix is a huge problem that cannot be resolved without much experiments trial and errors. like #18572
So I'd not expect anything about simplifying

And this is the reason why I had suggested other ways like not doing any multiplications
https://github.com/sympy/sympy/pull/18565#issuecomment-582620561
But if this is difficult to resolve now, I'd keep this not complicated for now